beacon-chain/rpc/eth/rewards/handlers.go

```diff
@@ -166,22 +178,318 @@ func (s *Server) BlockRewards(w http.ResponseWriter, r *http.Request) {
 	network.WriteJson(w, response)
 }
 
+// TODO: Explain the flow
+func (s *Server) AttestationRewards(w http.ResponseWriter, r *http.Request) {
+	segments := strings.Split(r.URL.Path, "/")
+	epoch, err := strconv.ParseUint(segments[len(segments)-1], 10, 64)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not decode epoch: " + err.Error(),
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	if primitives.Epoch(epoch) < params.BeaconConfig().AltairForkEpoch {
+		errJson := &network.DefaultErrorJson{
+			Message: "Attestation rewards are not supported for Phase 0",
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	currentEpoch := uint64(slots.ToEpoch(s.TimeFetcher.CurrentSlot()))
+	if epoch >= currentEpoch-1 {
+		errJson := &network.DefaultErrorJson{
+			Code:    http.StatusBadRequest,
+			Message: "Attestation rewards are available after two epoch transitions to ensure all attestation have a chance of inclusion",
+		}
```

Comment:
```suggestion
	if primitives.Epoch(epoch) < params.BeaconConfig().AltairForkEpoch {
		errJson := &network.DefaultErrorJson{
			Message: "Attestation rewards are not supported for Phase 0",
			Code:    http.StatusNotFound,
		}
		network.WriteError(w, errJson)
		return
	}
	currentEpoch := uint64(slots.ToEpoch(s.TimeFetcher.CurrentSlot()))
	if epoch >= currentEpoch-1 {
		errJson := &network.DefaultErrorJson{
			Code:    http.StatusNotFound,
			Message: "Attestation rewards are available after two epoch transitions to ensure all attestation have a chance of inclusion",
		}
```

> If epoch is before Altar, return 404.
> If epoch is current or previous epoch, return 404.

Looks like the return status code does not satisfy the spec in your description.

Reply:
Thanks for noticing. I made a mistake in the description.